texastribune/salesforce-stripe
# Add a campaignID query param

#### What's this PR do?

Adds a `campaignId` query parameter that goes into Salesforce. 

#### Why are we doing this? How does it help us?

It allows marketing to track where users came from and what campaign was used. 

#### How should this be manually tested?

Start the server and celery and then try the following URLs:

- http://local.texastribune.org/memberform?campaignId=blah&amount=85
- http://local.texastribune.org/memberform?campaignId=701J0000002BZ1P&amount=85
- http://local.texastribune.org/donateform?campaignId=blah&amount=85&installmentPeriod=monthly
- http://local.texastribune.org/donateform?campaignId=701J0000002BZ1P&amount=85&installmentPeriod=monthly
- http://local.texastribune.org/circleform?&amount=84&installmentPeriod=monthly&installments=36&campaignId=blah
- http://local.texastribune.org/circleform?&amount=84&installmentPeriod=monthly&installments=36&campaignId=701J0000002BZ1P
- http://local.texastribune.org/blastform?campaignId=blah
- http://local.texastribune.org/blastform?campaignId=701J0000002BZ1P

Confirm in each case that a Salesforce entry gets created and that in of the `blah` campaignIds that there's a note in the server log about it retrying because the ID was invalid. You can also try a few URLs that don't have that query param at all and make sure it works.

#### Have automated tests been added?

No, but a few have been fixed.

#### Has this been tested on mobile?

N/A

#### Are there performance implications?

No.

#### What are the relevant tickets?

- https://3.basecamp.com/3098728/buckets/736178/todos/638458285
- https://3.basecamp.com/3098728/buckets/736178/todos/638458406

#### How should this change be communicated to end users?

Let Rebecca know that this is complete.

#### Next steps?

No.

#### Smells?

None that I can think of.

#### Has the relevant documentation/wiki been updated?

I put an example of using the campaignId here: 

https://wiki.texastribune.org/display/ERP/Stripe+implementation+project

#### Technical debt note

None.
